### ISSUE TITLE
Needless message printed with --failed-first and no failed tests

### ISSUE THREAD

**Issue by nedbat** [pytest-dev/pytest#3853]
``--failed-first`` always means, run all the tests, but order the recent failure to run first.  But if there were no failures, it always prints a message that it is going to run all the tests. This message is not needed.
```
$ cat test_it.py
def test_it():
    assert 1 == 1

$ pytest
================== test session starts ===================
platform darwin -- Python 2.7.14, pytest-3.7.1, py-1.5.4, pluggy-0.7.1
rootdir: /private/tmp/failedfirst, inifile:
plugins: xdist-1.22.5, forked-0.2
collected 1 item

test_it.py .                                                                                                                                                               [100%]

================ 1 passed in 0.01 seconds ================

$ pytest --failed-first
================== test session starts ===================
platform darwin -- Python 2.7.14, pytest-3.7.1, py-1.5.4, pluggy-0.7.1
rootdir: /private/tmp/failedfirst, inifile:
plugins: xdist-1.22.5, forked-0.2
collected 1 item
run-last-failure: run all (no recorded failures)

test_it.py .                                                                                                                                                               [100%]

================ 1 passed in 0.01 seconds ================

$ pytest -q --failed-first
run-last-failure: run all (no recorded failures)
.                                                                                                                                                                          [100%]
1 passed in 0.01 seconds

```

**Comment by nicoddemus** [comment]
Hi @nedbat,

I think the idea is to make sure the plugin is actually working... I personally like to see the message to make sure there are actually no failures and now all tests are running.

**Comment by nedbat** [comment]
Yes :)

**Comment by nicoddemus** [comment]
#3886 deals with handling `-q` properly, after that gets merged we can them suppress the output when there are no recorded failures on the `features` branch.

**Comment by dhirensr** [comment]
@nedbat @nicoddemus  : I have read the complete issue and also #3886 ! does this ticket say that when --lf ,then the output now is "run-last-failure: run all (no recorded failures)" but instead it should behave exactly as --q and nothing should display?

**Comment by nicoddemus** [comment]
@dhirensr exactly! 😁

**Comment by dhirensr** [comment]
@nicoddemus : where do we change the code like do we send mode as empty string or change the print function ? could you also guide me where is the print ignored in case of --quiet ,I am not able to find that piece of code which ignores the print statement and prints nothing!

**Comment by dhirensr** [comment]
@nicoddemus : Oh yeah! that i had already fixed! I just thought I need to remove "platform linux -- Python 3.5.2, pytest-unknown, py-1.6.0, pluggy-0.7.1
rootdir: /home/dhirensr/pytest, inifile: tox.ini
collected 2 items                                                                                                                                                                                          
" also! 
I thought like complete terminal should look empty that's the requirement!
I am making a PR with test! I need to mention in the document also right? Thanks a lot for helping man!

**Comment by dhirensr** [comment]
@nicoddemus : PR submitted! https://github.com/pytest-dev/pytest/pull/3912
Please review and let me know if any changes required! I changed the code ,added a test and also removed 1 test option which was checking "run all" string.
Thanks a lot for helping! :smile:

**Comment by nicoddemus** [comment]
Sure thing, thanks @dhirensr, I will take a look as soon as I get a chance! 👍

**Comment by nicoddemus** [comment]
The complete feature will be available in `pytest-3.8`. Thanks @nedbat!

**Comment by nicoddemus** [comment]
And thanks @dhirensr for the PR! 🤗
